internal/webhook/external/server.go

```diff
@@ -32,8 +35,18 @@ func (s *server) Serve(ctx context.Context, l net.Listener) error {
 	logger := logging.LoggerFromContext(ctx)
 	mux := http.NewServeMux()
 
-	// TODO: Register handlers here
-	// mux.Handle("/", ...)
+	ghProvider, err := github.NewProvider()
+	if err != nil {
+		return fmt.Errorf("failed to initialize github provider: %w", err)
+	}
+
+	mux.Handle("POST /v1/github",
+		handlers.NewRefreshWarehouseWebhook(
+			ghProvider,
+			s.client,
+		),
+	)
+	// TODO(fuskovic): support additional providers
```

Comment:
See [this comment](https://github.com/akuity/kargo/pull/4000/files#r2080510829) for more insight into why we're probably, eventually going to have just one single endpoint listening to `/`.

Again... nothing actionable here just yet.